### PR TITLE
I/7802: Update the `Image` feature and API docs for `Insert image via URL` feature.

### LINKED ISSUE #7802: Update Image documentation for inserting via URL
## 📝 Provide a description of the new feature

- Update feature documentation with demo and a new section about inserting images via URL.
- Update API / config documentation for the `Image` feature with how to configure the `ImageUploadUI` panel view.

### PR DESCRIPTION
**Pull request by panr**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Docs (image): Updated feature and API documentation for the **Insert image via URL** feature. Closes #7802.

Docs (image): Added warning that shows CK error when the integration component view is missing.

### Additional information:
- This PR depends on `i/7794`.

### CODE REVIEW

File: packages/ckeditor5-image/docs/features/image.md [hunk 1]
```diff
@@ -82,6 +82,97 @@ By default, if the image caption is empty, the `<figcaption>` element is not vis
 
 See the {@link features/image-upload Image upload} guide.
 
+## Insert image via URL
+
+Another great feature is an ability to insert an image via URL address. The feature is available under the dropdown of the **Image Upload** feature. When you click on the arrow button, you will see a small panel view containing the form for an image source URL. Once you fill it, you will be able to insert the image into the editor's content.
```
Comment (godai78) [inline packages/ckeditor5-image/docs/features/image.md:87]:
```suggestion
Another great feature is the ability to insert an image via a URL address. This feature is available under the **Image Upload** feature dropdown. When you click on the arrow button, you will see a small panel view containing the form for an image source URL. Once you fill it, you will be able to insert the image into the editor's content.
```

File: packages/ckeditor5-image/docs/features/image.md [hunk 2]
```diff
@@ -82,6 +82,97 @@ By default, if the image caption is empty, the `<figcaption>` element is not vis
...
+
+### Replace image via URL
+
+Yet, another cool thing that the feature introduces is an ability to replace previously inserted image without loosing its styling and custom attributes (if it has ones). When you click on image and open the `{@link module:image/imageupload/ui/imageuploadpanelview~ImageUploadPanelView#dropdownView ImageUpload dropdown}` you will see that the **Insert image via url** form is already filled. This is the value of the selected image. You can easily edit it and update through the form.
```
Comment (godai78) [inline packages/ckeditor5-image/docs/features/image.md:91]:
```suggestion
Yet another cool function the feature introduces is the ability to replace a previously inserted image without loosing its styling and custom attributes (if it has ones). If you click on the image and open the `{@link module:image/imageupload/ui/imageuploadpanelview~ImageUploadPanelView#dropdownView ImageUpload dropdown}` you will see that the **Insert image via url** form is already filled. This is the URL of the current image. You can easily edit it and update it through the form.
```

File: packages/ckeditor5-image/docs/features/image.md [hunk 3]
```diff
@@ -82,6 +82,97 @@ By default, if the image caption is empty, the `<figcaption>` element is not vis
...
+
+<info-box>
+<strong>Bonus tip</strong>: You can even edit the URL of an image inserted from the native file system or from — for example — `{@link module:ckfinder/ckfinder~CKFinder}`.
+</info-box>
+
+Here's the short demo:
```
Comment (godai78) [inline packages/ckeditor5-image/docs/features/image.md:97]:
```suggestion
Check it out in this short demo:
```

File: packages/ckeditor5-image/docs/features/image.md [hunk 4]
```diff
@@ -82,6 +82,97 @@ By default, if the image caption is empty, the `<figcaption>` element is not vis
...
+
+{@snippet features/image-insert-via-url}
+
+### How to add integrations
+
+By default, this feature doesn't need any configuration. You will get a dropdown button, where the main action (icon button) will open a native file dialog and the arrow button will open a dropdown with **Insert image via url** feature.
+
+#### CKFinder
+
+We also support the integration with {@link module:ckfinder/ckfinder~CKFinder} in that dropdown, but that's something that needs to be configured.
```
Comment (godai78) [inline packages/ckeditor5-image/docs/features/image.md:107]:
```suggestion
The integration with {@link module:ckfinder/ckfinder~CKFinder} in that dropdown is also supported, but it needs to be configured.
```

File: packages/ckeditor5-image/docs/features/image.md [hunk 5]
```diff
@@ -82,6 +82,97 @@ By default, if the image caption is empty, the `<figcaption>` element is not vis
...
+
+Here's how you can do this:
```
Comment (godai78) [inline packages/ckeditor5-image/docs/features/image.md:109]:
```suggestion
Use this method to do this:
```

File: packages/ckeditor5-image/docs/features/image.md [hunk 6]
```diff
@@ -82,6 +82,97 @@ By default, if the image caption is empty, the `<figcaption>` element is not vis
...
+				}
+			}
+		},
+		ckfinder: {
+			uploadUrl: ... // CKFinder upload URL.
+		}
+	})
+```
+
+As you can see, we added here a new `{@link module:image/imageupload~ImageUploadPanelConfig#items image.upload.panel.items}` configuration, where we defined our desired built-in integrations. Both strings (`insertImageViaUrl` and `openCKFinder`) are special tokens that `{@link module:image/imageupload~ImageUpload}` plugin understands.
```
Comment (godai78) [inline packages/ckeditor5-image/docs/features/image.md:133]:
```suggestion
Observe that a new `{@link module:image/imageupload~ImageUploadPanelConfig#items image.upload.panel.items}` configuration defines the desired built-in integrations. Both strings (`insertImageViaUrl` and `openCKFinder`) are special tokens that the `{@link module:image/imageupload~ImageUpload}` plugin understands.
```

File: packages/ckeditor5-image/docs/features/image.md [hunk 7]
```diff
@@ -82,6 +82,97 @@ By default, if the image caption is empty, the `<figcaption>` element is not vis
...
+				}
+			}
+		}
+	} )
+	.then( ... )
+	.catch( ... );
+```
+
+<info-box>
+Keep in mind that the order of the integrations in configuration equals the order of those views in the `{@link module:image/imageupload/ui/imageuploadpanelview~ImageUploadPanelView#dropdownView}`. So in the dropdown panel view you will have: **Insert image via URL** form, next **CKFinder** button and then your custom component for **PluginX**.
```
Comment (godai78) [inline packages/ckeditor5-image/docs/features/image.md:171]:
```suggestion
Keep in mind that the order of the integrations in the configuration affects the options order in the `{@link module:image/imageupload/ui/imageuploadpanelview~ImageUploadPanelView#dropdownView}`. Therefore you will have: the **Insert image via URL** form, next the **CKFinder** button and then your custom component for **PluginX** in the dropdown panel view.
```

File: packages/ckeditor5-image/docs/features/image.md [hunk 8]
```diff
@@ -82,6 +82,97 @@ By default, if the image caption is empty, the `<figcaption>` element is not vis
...
+
+You can always change that order if you need.
```
Comment (godai78) [inline packages/ckeditor5-image/docs/features/image.md:173]:
```suggestion
You can always change that order if you need by changing the order of the integrations.
```

Comment (godai78) [reply packages/ckeditor5-image/docs/features/image.md:173]: I hope this is phrased correctly in term of technical names.

File: packages/ckeditor5-image/src/imageupload.js [hunk 9]
```diff
@@ -83,3 +83,55 @@ export default class ImageUpload extends Plugin {
...
+ * See {@link module:core/editor/editorconfig~EditorConfig all editor options}.
+ *
+ * @interface module:image/imageupload~ImageUploadPanelConfig
+ */
+
+/**
+ * The list of {@link module:image/imageupload~ImageUpload} integrations.
+ *
+ * The option accepts string tokens.
+ * * for predefined integrations, we have two special strings: `insertImageViaUrl` and `openCKFinder`.
```
Comment (godai78) [inline packages/ckeditor5-image/src/imageupload.js:116]:
```suggestion
 * * for predefined integrations, there are two special strings: `insertImageViaUrl` and `openCKFinder`.
```

File: packages/ckeditor5-image/src/imageupload.js [hunk 10]
```diff
@@ -83,3 +83,55 @@ export default class ImageUpload extends Plugin {
...
+ * The former adds **Insert image via URL** feature, but the latter adds built-in **CKFinder** integration.
```
Comment (godai78) [inline packages/ckeditor5-image/src/imageupload.js:117]:
```suggestion
 * The former adds **Insert image via URL** feature, while the latter adds built-in **CKFinder** integration.
```

**Comment by panr** [comment]
Thanks @godai78 for the review 👍

**Comment by Reinmar** [comment]
TODO: Remove the part about integrations and the CKFinder integration from the demo. We'll work on that in the next iteration.

**Comment by panr** [comment]
The `integrations` section has been removed and mentioned here https://github.com/ckeditor/ckeditor5/issues/7875#issuecomment-676085499.

**Comment by panr** [comment]
@Reinmar just updated the docs as well.

**Pull request merged**